qt/scientific_interfaces/Indirect/test/IndirectDataTablePresenterTest.h

```diff
@@ -0,0 +1,368 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#ifndef MANTIDQT_INDIRECTDATATABLEPRESENTERTEST_H_
+#define MANTIDQT_INDIRECTDATATABLEPRESENTERTEST_H_
+
+#include <cxxtest/TestSuite.h>
+#include <gmock/gmock.h>
+
+#include "IndirectDataTablePresenter.h"
+#include "IndirectFittingModel.h"
+
+#include "MantidAPI/FrameworkManager.h"
+#include "MantidKernel/WarningSuppressions.h"
+#include "MantidTestHelpers/IndirectFitDataCreationHelper.h"
+
+using namespace Mantid::API;
+using namespace Mantid::IndirectFitDataCreationHelper;
+using namespace MantidQt::CustomInterfaces::IDA;
+using namespace testing;
+
+GNU_DIAG_OFF_SUGGEST_OVERRIDE
+
+/// Mock object to mock the model
+class MockIndirectDataTableModel : public IndirectFittingModel {
+public:
+  /// Public methods
+  MOCK_CONST_METHOD2(getFittingRange,
+                     std::pair<double, double>(std::size_t dataIndex,
+                                               std::size_t spectrum));
+  MOCK_CONST_METHOD2(getExcludeRegion,
+                     std::string(std::size_t dataIndex, std::size_t index));
+  MOCK_CONST_METHOD0(isMultiFit, bool());
+  MOCK_CONST_METHOD0(numberOfWorkspaces, std::size_t());
+
+  MOCK_METHOD3(setStartX, void(double startX, std::size_t dataIndex,
+                               std::size_t spectrum));
+  MOCK_METHOD3(setEndX,
+               void(double endX, std::size_t dataIndex, std::size_t spectrum));
+  MOCK_METHOD3(setExcludeRegion,
+               void(std::string const &exclude, std::size_t dataIndex,
+                    std::size_t spectrum));
+
+private:
+  std::string sequentialFitOutputName() const override { return ""; };
+  std::string simultaneousFitOutputName() const override { return ""; };
+  std::string singleFitOutputName(std::size_t index,
+                                  std::size_t spectrum) const override {
+    UNUSED_ARG(index);
+    UNUSED_ARG(spectrum);
+    return "";
+  };
+
+  std::vector<std::string> getSpectrumDependentAttributes() const override {
+    return {};
+  };
+};
+
+GNU_DIAG_ON_SUGGEST_OVERRIDE
+
+class IndirectDataTablePresenterTest : public CxxTest::TestSuite {
+public:
+  /// Needed to make sure everything is initialized
+  IndirectDataTablePresenterTest() { FrameworkManager::Instance(); }
+
+  static IndirectDataTablePresenterTest *createSuite() {
+    return new IndirectDataTablePresenterTest();
+  }
+
+  static void destroySuite(IndirectDataTablePresenterTest *suite) {
+    delete suite;
+  }
+
+  void setUp() override {
+    m_model = std::make_unique<NiceMock<MockIndirectDataTableModel>>();
+    createEmptyTableWidget(5, 5);
+    m_presenter = std::make_unique<IndirectDataTablePresenter>(
+        std::move(m_model.get()), std::move(m_table.get()));
```

Comment:
Where is m_table initialised?

Reply:
m_table is initialized within the `createEmptyTableWidget` function. I believe this is in the private section of the `IndirectDataTablePresenterTest` class

Reply:
Long term it could be more sustainable to assign m_table to createEmptyTableWidget created in a helper file. This could make it a bit more readable. Up to you though.

Reply:
I agree this isn't very readable, so I have made it so `createEmptyTableWidget` returns the table for assignment to m_table. I want to avoid putting this in a helper because including `<QTableWidget>` in the helper causes problems.